### ISSUE TITLE
Inpainting only does the first image

### ISSUE THREAD

**Issue by landmann** [huggingface/diffusers#4854]
### Describe the bug

I'm trying to inpaint 4 different images, but I only get the first one repeated 4 times.

I believe the error is in this file src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

I did a bit of digging and it seems `prepare_mask_and_maked_image` works fine, masking all four images individually, so the error is somewhere down the pipeline. My intuition tells me the first image is being copied 4 times somewhere in the code.

### Result of images after mask is applied
![image](https://github.com/huggingface/diffusers/assets/12756081/27daf6eb-be39-4783-ae28-987265dc3f46)

However, the output is still just the first image repeated four times.

### Reproduction

        recreated = self.inpaint_pipe(
            prompt=[params.prompt]*4, 
            negative_prompt=[params.negative_prompt]*4,
            image=cropped_images,
            mask_image=masks, 
            num_inference_steps=params.num_inference_steps, 
            guidance_scale=params.guidance_scale, # adhere to prompt
            strength=0.7,
            num_images_per_prompt=1,
            width=cropped_images[0].width,
            height=cropped_images[0].height,
        ).images

### System Info

None

### Who can help?

@patrickvonplaten @sayakpaul

**Comment by patrickvonplaten** [comment]
cc @yiyixuxu can you check this?

**Comment by landmann** [comment]
@yiyixuxu guys did you find anything? Or any ideas why this could be happening?

**Comment by yiyixuxu** [comment]
@landmann 

Happy to look into it! 

Could you provide a self-contained, reproducible example? The example you provided is not reproducible since it does not include the inputs you used.

You can read here about[ how to make a reproducible minimal bug report](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#21-reproducible-minimal-bug-reports). Also see below an example of a reproducible code snippet. It will really help us out :)

Thanks
YiYi


```python
import PIL
import requests
import torch
from io import BytesIO

from diffusers import StableDiffusionInpaintPipeline


def download_image(url):
    response = requests.get(url)
    return PIL.Image.open(BytesIO(response.content)).convert("RGB")


img_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo.png"
mask_url = "https://raw.githubusercontent.com/CompVis/latent-diffusion/main/data/inpainting_examples/overture-creations-5sI6fQgYIuo_mask.png"

init_image = download_image(img_url).resize((512, 512))
mask_image = download_image(mask_url).resize((512, 512))

pipe = StableDiffusionInpaintPipeline.from_pretrained(
    "runwayml/stable-diffusion-inpainting",
    revision="fp16",
    torch_dtype=torch.float16,
)
pipe = pipe.to("cuda")

prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
image = pipe(prompt=prompt, image=init_image, mask_image=mask_image).images[0]
```

**Comment by landmann** [comment]
Hmm, seems like I made a mistake myself 😭 Apologies for the hassle, but thank you so much for the responsiveness, team!!!

**Comment by landmann** [comment]
I thought it was fixed, but now I found the issue... it seems like the first image is being used as the background even though the masks are being used accordingly.  Surprisingly, it only malfunctions when I load the Realistic Vision model - not when I use the `stable-diffusion-inpainting`. I thought loading the model would just change the weights, not the process?

```
    def run_test(self, params: GenParams):
        import PIL
        import requests
        import torch
        from io import BytesIO

        from diffusers import StableDiffusionInpaintPipeline


        def download_image(url):
            response = requests.get(url)
            return PIL.Image.open(BytesIO(response.content)).convert("RGB")


        imgs = []
        masks = []
        
        for img_url, mask_url in [
            ("https://i.ibb.co/DM9qbW8/x0.png","https://i.ibb.co/4TTHp6w/xxx.png"),
            ("https://i.ibb.co/J7DXFmB/x1.png","https://i.ibb.co/2gkxyWB/x1.png"),
            ("https://i.ibb.co/8gRgDYG/x3.png","https://i.ibb.co/WvZ7Q5h/x2.png"),
            ("https://i.ibb.co/KXVWT5D/x4.png","https://i.ibb.co/P5rTZFQ/x3.png")]:

            imgs.append(download_image(img_url))
            masks.append(download_image(mask_url))

        model_path = MODEL_CACHE_FOLDER+"/SG161222/Realistic_Vision_V5.1_noVAE"

        pipe = StableDiffusionInpaintPipeline.from_pretrained(
            model_path,
            revision="fp16",
            torch_dtype=torch.float16,
        )
        pipe = pipe.to("cuda")

        prompt = "Face of a yellow cat, high resolution, sitting on a park bench"

        return pipe(
            prompt=[prompt]*4, 
            image=imgs, 
            mask_image=masks,
            width=512,
            height=512
        ).images
        
```

**Comment by yiyixuxu** [comment]
@landmann oh thank you! - looking into it now:)

**Comment by yiyixuxu** [comment]
@landmann 
thanks so much for the bug report! it's super helpful. It is indeed a bug in all our inpainting pipelines. I included more details here https://github.com/huggingface/diffusers/pull/5033

**Comment by landmann** [comment]
@yiyixuxu my pleasure! Was this merged to main? I think I still get the error..

**Comment by patrickvonplaten** [comment]
It should be merged and fixed now. @landmann can you maybe open a new issue in case you still get errors?
